BrowserStack.js

```diff
@@ -1,22 +1,36 @@
-var webdriver = require('browserstack-webdriver');
+const { Builder, By, Key, until } = require('selenium-webdriver');
+const http = require('http');
 
-var capabilities = {
-	'browserName' : 'firefox', 
-	'browserstack.user' : 'USERNAME',
-	'browserstack.key' : 'ACCESS_KEY'
-}
-var driver = new webdriver.Builder().
-	usingServer('http://hub.browserstack.com/wd/hub').
-	withCapabilities(capabilities).
-	build();
+const BROWSERSTACK_USERNAME = process.env.BROWSERSTACK_USERNAME || 'BROWSERSTACK_USERNAME';
+const BROWSERSTACK_ACCESS_KEY = process.env.BROWSERSTACK_ACCESS_KEY || 'BROWSERSTACK_ACCESS_KEY';
 
-driver.get('http://www.google.com');
-driver.findElement(webdriver.By.name('q')).sendKeys('BrowserStack');
-driver.findElement(webdriver.By.name('btnG')).click();
-driver.wait(function() {
-	return driver.getTitle().then(function(title) {
-		return title === 'BrowserStack - Google Search';
+let HttpAgent = new http.Agent({
+	keepAlive: true,
+});
+
+let capabilities = {
+	browserName: 'Firefox',
+	name: 'Firefox Test',
+	os: 'Windows',
+	build: 'Test Build 01',
+	project: 'My Awesome App',
+	'browserstack.debug': true,
+};
+
+let driver = new Builder()
+	.usingHttpAgent(HttpAgent)
+	.withCapabilities(capabilities)
+	.usingServer(`http://${BROWSERSTACK_USERNAME}:${BROWSERSTACK_ACCESS_KEY}@hub-cloud.browserstack.com/wd/hub`)
+	.build();
+
+driver.get('http://www.google.com/ncr').then(() => {
+	driver.findElement(By.name('q')).then((element) => {
+		element.sendKeys('BrowserStack', Key.RETURN).then(() => {
+			driver.wait(until.titleContains('BrowserStack')).then(() => driver.getTitle().then((title) => {
+				console.log(title);
+				driver.quit();
+			}));
+		})
 	});
-}, 1000);
+});
 
```

Comment:
for now lets go ahead with this. need to figure out better handling of finally with chained promises.